File: easy/1295_find_numbers_with_even_number_of_digits/solution.py
from typing import List


class Solution:
    def findNumbers(self, nums: List[int]) -> int:
        def countDigit(num: int) -> int:
            digitCount = 0

            while num:
                digitCount += 1
                num //= 10

            return digitCount

        numCount = 0

        for num in nums:
            if countDigit(num) % 2 == 0:
                numCount += 1

        return numCount

File: easy/1295_find_numbers_with_even_number_of_digits/solution.cpp
#include <vector>

class Solution {
public:
  int findNumbers(std::vector<int> &nums) {
    auto countDigit = [](int num) {
      int digitCount = 0;

      while (num) {
        ++digitCount;
        num /= 10;
      }

      return digitCount;
    };

    int numCount = 0;

    for (const auto &num : nums) {
      if (countDigit(num) % 2 == 0)
        ++numCount;
    }

    return numCount;
  }
};
